### PR TITLE
Consistent use of `.out` for test expectations

### PR DESCRIPTION
**Pull request by sbc100**
We previously accepted either `.out` or `.txt` but for no
good reason that I could determine.

### CODE REVIEW

File: tests/runner.py [hunk 1]
```diff
@@ -1068,23 +1068,17 @@ def do_run_from_file(self, filename, expected_output, **kwargs):
...
-                         .format(test_path, ext_list))
-      assert count <= 1, ("Test file {} found with multiple valid extensions {}"
-                          .format(test_path, ext_list))
-      return ret
-
-    srcfile = find_files('.c', '.cpp')
-    outfile = find_files('.out', '.txt')
+    srcfile = path_from_root(*path)
+
+    # TODO(sbc): Have tests explciictly pass filename
```
Comment (kripken) [inline tests/runner.py:1073]:
```suggestion
    # TODO(sbc): Have tests explicitly pass filename
```

**Comment by sbc100** [comment]
> I would prefer `.txt` actually since these are text files. But probably not worth the effort to switch, lgtm.

But if you see a `.txt` file it could be anything right?   I kind of like seeing the `.out` because it tells me what it is.   I mean `.cpp` files are also text right :)

I'm not completely convinced I'm correct thought... @dschuff  what do you think?

**Comment by sbc100** [comment]
I also chose `.out` since there were more existing tests using that style.

**Comment by kripken** [comment]
Fair point, `out` does indicate what it is in a useful way.

File: tests/runner.py (hunk 1 above)
Comment (sbc100) [reply tests/runner.py:1073]: Oops forgot to fix this before landing... I'm hoping to remove the TODO shortly though... so meh

**Pull request merged**